src/keys.rs

```diff
@@ -0,0 +1,371 @@
+                "Bls({:<8})",
+                HexFmt(&pub_key.to_bytes()[..XOR_NAME_LEN])
+            ),
+            Self::BlsShare(pub_key) => write!(
+                formatter,
+                "BlsShare({:<8})",
+                HexFmt(&pub_key.to_bytes()[..XOR_NAME_LEN])
+            ),
+        }
+    }
+}
+
+impl Display for PublicKey {
+    fn fmt(&self, formatter: &mut Formatter) -> fmt::Result {
+        Debug::fmt(self, formatter)
+    }
+}
+
+/// Wrapper type for different secret key types.
+#[derive(Deserialize)]
+pub enum SecretKey {
+    /// Ed25519 secret key.
+    Ed25519(ed25519_dalek::SecretKey),
+    /// BLS secret key.
+    Bls(threshold_crypto::SecretKey),
+    /// BLS secret key share.
+    BlsShare(threshold_crypto::SecretKeyShare),
+}
+
+// Need to manually implement this due to a missing impl in `Ed25519::SecretKey`.
+impl Clone for SecretKey {
+    fn clone(&self) -> Self {
+        match self {
+            Self::Ed25519(sec_key) => Self::Ed25519(unwrap!(ed25519_dalek::SecretKey::from_bytes(
+                &sec_key.to_bytes()
+            ))),
+            Self::Bls(sec_key) => Self::Bls(sec_key.clone()),
+            Self::BlsShare(sec_key) => Self::BlsShare(sec_key.clone()),
+        }
+    }
+}
+
+// Need to manually implement this due to a missing impl in `Ed25519::SecretKey`.
+impl PartialEq for SecretKey {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Ed25519(sec_key), Self::Ed25519(other_sec_key)) => {
+                sec_key.to_bytes() == other_sec_key.to_bytes()
+            }
+            (Self::Bls(sec_key), Self::Bls(other_sec_key)) => sec_key == other_sec_key,
+            (Self::BlsShare(sec_key), Self::BlsShare(other_sec_key)) => sec_key == other_sec_key,
+            _ => false,
+        }
+    }
+}
+
+// Need to manually implement this due to a missing impl in `Ed25519::SecretKey`.
+impl Eq for SecretKey {}
+
+// Need to manually implement this due to a missing impl in `threshold_crypto::SecretKey`.
+impl Serialize for SecretKey {
+    fn serialize<S>(&self, serializer: S) -> ::std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            Self::Ed25519(sec_key) => {
+                serializer.serialize_newtype_variant("SecretKey", 0, "Ed25519", &sec_key)
+            }
+            Self::Bls(sec_key) => {
+                serializer.serialize_newtype_variant("SecretKey", 1, "Bls", &SerdeSecret(&sec_key))
+            }
+            Self::BlsShare(sec_key) => serializer.serialize_newtype_variant(
+                "SecretKey",
+                2,
+                "BlsShare",
+                &SerdeSecret(&sec_key),
+            ),
+        }
+    }
+}
+
+impl SecretKey {
+    /// Generates a random BLS key.
+    #[cfg(test)]
+    pub fn random_bls() -> Self {
+        Self::Bls(threshold_crypto::SecretKey::random())
+    }
+
+    // TODO: implement zbase32 encoding and decoding?
```

Comment:
We might need this since a user may like to encode his secret key and store it as a string somewhere locally.